File: AUGUST-2025/09-08-2025/p231.php
class Solution {

    /**
     * @param Integer $n
     * @return Boolean
     */
    function isPowerOfTwo($n) {
        if ($n < 0) return false;
        $absN = abs($n);
        if ($absN <= 1) {
            if ($n == 0) return false;
            return true;
        }
        $n = $absN;

        while ($n > 1) {
            if ($n % 2 != 0) {
                return false;
            }
            $n /= 2;
        }
        return true;
    }
}

File: AUGUST-2025/09-08-2025/p231.swift
class Solution {
    func isPowerOfTwo(_ n: Int) -> Bool {
        var nCont = n;

        guard nCont > 0 else {
            return false;
        }

        guard nCont > 1 else {
            return true;
        }

        while (nCont >= 1) {
            if (nCont % 2 == 0) {
                nCont /= 2;

                if (nCont == 1) {
                    return true;
                }
            } else {
                return false;
            }
        }

        return false;
    }
}